contracts/HenkakuBadge.sol

```diff
@@ -1,33 +1,53 @@
-//SPDX-License-Identifier: Unlicense
+//SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
 import "hardhat/console.sol";
 import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+
+contract HenkakuBadge is ERC1155, Ownable {
+    using Counters for Counters.Counter;
+    Counters.Counter private _tokenIds;
 
-contract HenkakuBadge is ERC1155 {
     struct Badge {
         bool mintable;
         bool transerable;
         uint256 amount;
         string tokenURI;
     }
 
-    uint256 tokenId;
-    mapping(uint256 => Badge) bages;
+    mapping(uint256 => Badge) public badges;
 
     constructor() ERC1155("") {}
 
-    function badges() public view returns (Badge[] memory) {}
+    function getBadges() public view returns (Badge[] memory) {}
 
     function badgeOf(address _of) public returns (Badge[] memory) {}
 
-    // only by owner
-    function createBadge() public {
-        // it can be setBadge Attribute
+    event NewBadge(uint256 indexed id, bool mintable, uint256 amount);
+    event UpdateBadge(uint256 indexed id, bool mintable);
```

Comment:
It looks good, but I may not fully understand the reason for separating the events.

Reply:
hmmm, creation of newBadge and update of certain badge.
it should be seperated. or
`event UpdateBadge(..., bool initialized)`
sth like this would be ok to